### PR TITLE
Port SyncHandler to async/await

### PR DESCRIPTION
**Pull request by erikjohnston**

### CODE REVIEW

File: synapse/handlers/sync.py [hunk 1]
```diff
@@ -1143,8 +1133,7 @@ def _generate_sync_entry_for_device_list(
         else:
             return DeviceLists(changed=[], left=[])
 
-    @defer.inlineCallbacks
```
Comment (anoadragon453) [inline synapse/handlers/sync.py]: Missing an await here?

Comment (erikjohnston) [reply synapse/handlers/sync.py]: huh?

Comment (anoadragon453) [reply synapse/handlers/sync.py]: Hm, was on mobile and didn't see an `await` before `self.store.delete_messages_for_device`. Guess that's the last time I review on mobile :)

Comment (erikjohnston) [reply synapse/handlers/sync.py]: Ah, right, I was just confused by the fact it was on the function line. FTR `async def` doesn't need to have an `await` I don't think.

**Pull request merged**